src/NaturalSort/NaturalSortComparer.cs

```diff
@@ -0,0 +1,119 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace NaturalSort
+{
+    /// <summary>
+    /// Sorts strings "naturally", so that the string "2" shows up before the string "10".
+    /// </summary>
+    public class NaturalSortComparer : IComparer<string>
+    {
+        /// <summary>
+        /// A global instance of the comparer.
+        /// </summary>
+        public static readonly NaturalSortComparer Instance = new NaturalSortComparer();
+
+        static NaturalSortComparer()
+        {
+        }
+
+        private NaturalSortComparer()

```

Comment:
It didn't seem to make sense to be able to new up instances of this comparer, since it doesn't maintain any state, so I made it non-constructable and provided a static instance.